[ACR] Add SDK for Azure container registry

djyou opened the pull request:
SDK is generated from approved [swagger spec for Azure container registry](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-containerregistry/2016-06-27-preview/swagger/containerregistry.json).

djyou committed:
Add SDK for Azure container registry

azuresdkci commented:
Can one of the admins verify this patch?

azurecla commented:
Hi __@djyou__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (doyou). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

djyou commented:
/cc @SajayAntony @sivagms

hovsepm commented:
@azuresdkci add to whitelist

djyou commented:
Swagger link https://github.com/Azure/azure-rest-api-specs/blob/master/arm-containerregistry/2016-06-27-preview/swagger/containerregistry.json

djyou committed:
Add tests for 204 response in delete operation

shahabhijeet commented on src/ResourceManagement/ContainerRegistry/ContainerRegistry.Tests/Helpers/ContainerRegistryTestUtilities.cs:
```diff
@@ -0,0 +1,145 @@
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using Xunit;
+
+namespace ContainerRegistry.Tests
```
@djyou I don't see any tests that have been enabled, also I don't see anything in the logs for the tests being run.

djyou replied:
Is there another place we need to enable tests? Tests have been recorded in ResourceManagement/ContainerRegistry/ContainerRegistry.Tests/SessionRecords

markcowl replied:
tests are included

markcowl replied:
10 tests show as passed in the log for ContainerRegistry

markcowl commented on src/ResourceManagement/ContainerRegistry/ContainerRegistry.Tests/Helpers/ContainerRegistryTestUtilities.cs:
```diff
@@ -0,0 +1,145 @@
+{
+    public class ContainerRegistryTestUtilities
+    {
+        public static string DefaultLocation = "southcentralus";
```
Please do not encode a static default location.  It is best to query ARM for an appropriate location during test setup.  Using a static location causes maintenance problems when tests are run later - many environments do not contain this location, checking ARM for locations containing the needed RPs *always* works in every environment.

markcowl commented on src/ResourceManagement/ContainerRegistry/ContainerRegistry.Tests/project.json:
```diff
@@ -0,0 +1,42 @@
+      "version": "1.0.0"
+    },
+    "Microsoft.Azure.Test.HttpRecorder": "[1.6.7-preview,2.0.0)",
+    "Microsoft.Rest.ClientRuntime.Azure.TestFramework": "[1.3.5-preview,2.0.0)",
```
Please use latest versions of runtime and test framework

markcowl commented on src/ResourceManagement/ContainerRegistry/Microsoft.Azure.Management.ContainerRegistry/project.json:
```diff
@@ -0,0 +1,49 @@
+      "imports": [ "dnxcore50" ],
+      "dependencies": {
+        "System.Runtime.Serialization.Primitives": "4.1.1",
+        "Microsoft.Rest.ClientRuntime.Azure": "[3.3.1,4.0.0)"
```
Why not have a single runtime dependncy for all targets, since the dependency is the same.  Also, please update to the latest runtime version

markcowl reviewed (changes requested):
A few minor changes, otherwise LGTM

djyou committed:
PR review update

1. Update to the lastest package version.
2. Query default location from ARM.
3. Improved tests.

djyou committed:
Merge pull request #1 from djyou/doyou/review

PR review update

djyou commented:
@markcowl Thanks for you feedback. We have made requested changes.

djyou commented:
@markcowl Can we have this merged if there is no more comment?

SajayAntony commented:
:shipit:
